blog/_posts/2016-07-04-trait-method-performance.md

```diff
@@ -104,6 +104,15 @@ methods `should_inline`, `should_not_inline` and `try_to_inline`). A simplified
   - The inlining depth is limited (9 by default, `MaxInlineLevel`).
   - No inlining is performed if the callsite method is already very large.
 
+  // this is not entirelly true. It uses multiple measures of `large`.
+  // inlining is performed twice. Once, on bytecode and that's the part that you've correctly covered,
+  // second time on the assembly. And size limits are different and measure different notions.
+  // InlineSmallCode option guards inining of assembly.
+  // neither of approaches is good:
+  //  - the problem with counting size of bytecode before inlining is that it counts unreachable code&gotos
+  //  - the problem with inlining assembly happens as it frequently has ridiculosly bad code(eg a lot of moving data around) in "slow paths" and
+          the size of slow path contributes to the measured size of method.
+
```

Comment:
ok, interesting - i don't know / didn't read anything about what happens when a callsite is being compiled and the callee is already compiled. so this case is different?

what do you mean by "inlining is performed twice", is that C1 and C2? are C1 and C2 not using the same inlining heuristics? do you have references? links to the two implementations?

of course, "large" is very imprecise, but that's deliberate. i'm not trying to be precise here, just give an intuition. more precise descriptions are in linked references.